avalon/tools/workfiles/app.py

```diff
@@ -3,86 +3,120 @@
 import getpass
 import re
 import shutil
+import logging
+import platform
 
 from ...vendor.Qt import QtWidgets, QtCore
-from ... import style, io, api
+from ...vendor import qtawesome
+from ... import style, io, api, pipeline
 
 from .. import lib as tools_lib
+from ..widgets import AssetWidget
+from ..models import TasksModel
+
+log = logging.getLogger(__name__)
 
 module = sys.modules[__name__]
 module.window = None
 
 
 class NameWindow(QtWidgets.QDialog):
-    """Name Window"""
+    """Name Window to define a unique filename inside a root folder
 
-    def __init__(self, root):
-        super(NameWindow, self).__init__()
-        self.setWindowFlags(QtCore.Qt.FramelessWindowHint)
+    The filename will be based on the "workfile" template defined in the
+    project["config"]["template"].
 
-        self.result = None
-        self.setup(root)
-
-        self.layout = QtWidgets.QVBoxLayout()
-        self.setLayout(self.layout)
-
-        grid_layout = QtWidgets.QGridLayout()
-
-        label = QtWidgets.QLabel("Version:")
-        grid_layout.addWidget(label, 0, 0)
-        self.version_spinbox = QtWidgets.QSpinBox()
-        self.version_spinbox.setMinimum(1)
-        self.version_spinbox.setMaximum(9999)
-        self.version_checkbox = QtWidgets.QCheckBox("Next Available Version")
-        self.version_checkbox.setCheckState(QtCore.Qt.CheckState(2))
-        layout = QtWidgets.QHBoxLayout()
-        layout.addWidget(self.version_spinbox)
-        layout.addWidget(self.version_checkbox)
-        grid_layout.addLayout(layout, 0, 1)
-        # Since the version can be padded with "{version:0>4}" we only search
-        # for "{version".
-        if "{version" not in self.template:
-            label.setVisible(False)
-            self.version_spinbox.setVisible(False)
-            self.version_checkbox.setVisible(False)
+    """
 
-        label = QtWidgets.QLabel("Comment:")
-        grid_layout.addWidget(label, 1, 0)
-        self.comment_lineedit = QtWidgets.QLineEdit()
-        if "{comment}" not in self.template:
-            label.setVisible(False)
-            self.comment_lineedit.setVisible(False)
-        grid_layout.addWidget(self.comment_lineedit, 1, 1)
+    def __init__(self, parent, root, session=None):
+        super(NameWindow, self).__init__(parent=parent)
+        self.setWindowFlags(self.windowFlags() | QtCore.Qt.FramelessWindowHint)
 
-        grid_layout.addWidget(QtWidgets.QLabel("Preview:"), 2, 0)
-        self.label = QtWidgets.QLabel("File name")
-        grid_layout.addWidget(self.label, 2, 1)
+        self.result = None
+        self.host = api.registered_host()
+        self.root = root
+        self.work_file = None
+
+        if session is None:
+            # Fallback to active session
+            session = api.Session
```

Comment:
Fallback could be removed and `session` made a required argument for the Window since it's only used by the Work Files tool and it will always explicitly pass a session, or at least it should.